File: Algorytmy rozwiązywania gier/cell.py
from pieces import Piece


class Cell:
    def __init__(self, alias: str, piece=None):
        self.piece: Piece = piece
        self.alias: str = alias

    def __eq__(self, other) -> bool:
        return self.alias == other.alias

    def __repr__(self):
        if self.piece is None:
            return f'{self.alias}\nPiece: None'
        return f'{self.alias}\nPiece: {self.piece}'

    @property
    def marker(self):
        if self.piece is None:
            return ' '
        return self.piece.marker

File: Frącek Witold 254511 gry/Cell.kt


class Cell(val alias: String, var piece: Piece? = null) {

    val marker: String
        get() = piece?.marker ?: " "

    val isEmpty: Boolean
        get() = piece == null

    override fun equals(other: Any?): Boolean {
        if(other !is Cell){
            return false
        }
        return alias == other.alias
    }

    override fun hashCode(): Int {
        var result = alias.hashCode()
        result = 31 * result + (piece?.hashCode() ?: 0)
        return result
    }

    override fun toString(): String {
        return marker
    }
}